code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -540,6 +540,25 @@
 	if(volume >= 1)
 		T.wet(800, TURF_WET_LUBE)
 
+/datum/reagent/sodium_polyacrylate
+	name = "Sodium Polyacrylate"
+	id = SODIUM_POLYACRYLATE
+	description = "A super absorbent polymer that can absorb water based substances."
+	reagent_state = SOLID
+	color = "#FFFFFF"
+	density = 1.22
+	specheatcap = 4.14
+
+/datum/reagent/sodium_polyacrylate/reaction_turf(var/turf/simulated/T, var/volume)
+
+	if(..())
+		return 1
+
+	if(T.wet)
+		var/obj/effect/decal/cleanable/molten_item/I = new/obj/effect/decal/cleanable/molten_item(T)
```

Comment:
Needs to check if there isn't already one of these decals in turf so people can't make thousands of them per tile with smoke